main/main.cpp

```diff
@@ -2579,6 +2582,7 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 	GLOBAL_DEF_BASIC("xr/openxr/extensions/hand_tracking_controller_data_source", false); // XR_HAND_TRACKING_DATA_SOURCE_CONTROLLER_EXT
 	GLOBAL_DEF_RST_BASIC("xr/openxr/extensions/hand_interaction_profile", false);
 	GLOBAL_DEF_BASIC("xr/openxr/extensions/eye_gaze_interaction", false);
+#endif // _3D_DISABLED
```

Comment:
This should enclose the few commented out lines below which are also about XR.

Reply:
I've just done force push and I've just remembered why I didn't do it before. `TOOLS_ENABLED` means you compile editor. Doesn't editor must to have all the features or it can be compiled without xr?

Reply:
It's true that the editor can't be compiled with `disable_3d`, so it should be equivalent either way. But semantically I think it's clearer to exclude all XR related code in this case.

Reply:
There also exist the following part of the code which also have xr option inside `TOOLS_ENABLED`:

https://github.com/godotengine/godot/blob/0a9d8f04c10870c0f9f7bbd2e0505edc8494e299/main/main.cpp#L1045-L1059

But I though that editor in general must contain everything and is not supposed to be built without 3d?

Reply:
Hm, it could be:
```diff
diff --git a/main/main.cpp b/main/main.cpp
index 75797e31de..81bf5c735d 100644
--- a/main/main.cpp
+++ b/main/main.cpp
@@ -1055,8 +1055,11 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
                if (arg == "--audio-driver" ||
                                arg == "--display-driver" ||
                                arg == "--rendering-method" ||
-                               arg == "--rendering-driver" ||
-                               arg == "--xr-mode") {
+                               arg == "--rendering-driver"
+#ifndef _3D_DISABLED
+                               || arg == "--xr-mode"
+#endif
+               ) {
                        if (N) {
                                forwardable_cli_arguments[CLI_SCOPE_TOOL].push_back(arg);
                                forwardable_cli_arguments[CLI_SCOPE_TOOL].push_back(N->get());
```

but that's honestly a bit ugly, for something that's indeed not necessary as you pointed out.

It doesn't matter much either way, but I'd say in this case let's just live with the minor inconsistency and leave this one as is.